middleware.go

```diff
@@ -138,11 +140,30 @@ func (tw traceware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	ctx, span := tw.tracer.Start(
-		ctx, spanName,
+	// define span start options
+	spanOpts := []oteltrace.SpanStartOption{
 		oteltrace.WithAttributes(spanAttributes...),
 		oteltrace.WithSpanKind(oteltrace.SpanKindServer),
-	)
+	}
+
+	if tw.publicEndpointFn != nil && tw.publicEndpointFn(r) {
+		// mark span as the root span
+		spanOpts = append(spanOpts, oteltrace.WithNewRoot())
+
+		// linking incoming span context to the root span if possible
+		spanCtx := oteltrace.SpanContextFromContext(ctx)
+		if spanCtx.IsValid() && spanCtx.IsRemote() {
```

Comment:
can we add description here about what is `IsValid` and what is `IsRemote` for `spanCtx`? I need to check definition first to make sure what is it. Adding comments here could help us to know at the first hand.